### PR TITLE
Measure code coverage using GitLab CI

### PR DESCRIPTION
**Pull request by torkleyy**
Integrate GitLab CI + kcov + codecov.io

## Description

* Collect code coverage using `kcov`
* Upload coverage to codecov.io
* Use GitLab CI to do all the work
* Add badge to readme (not yet valid)

## PR Checklist

By placing an x in the boxes I certify that I have:

- [x] Ran `cargo test --all` locally if this modified any rs files.
- [x] Ran `cargo +stable fmt --all` locally if this modified any rs files.
- [x] Updated the content of the book if this PR would make the book outdated.
- [x] Added a changelog entry if this will impact users, or modified more than 5 lines of Rust that wasn't a doc comment.
- [x] Added unit tests for new APIs if any were added in this PR.
- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

### CODE REVIEW

**Comment by LucioFranco** [comment]
I dont even see this running on this PR or does this run only on master? 

cc @fhaynes what should we do about this? Have we made the choice to go with jenkins or go back to gitlab?

**Comment by torkleyy** [comment]
It _should_ only run on master ;)

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -44,3 +44,13 @@ run tests:
...
     - branches
     - merge_requests
+
+run kcov:
+  stage: test
+  image: docker:latest
+  services:
+    - docker:dind
+  script:
+    - docker --version
```
Comment (LucioFranco) [inline .gitlab-ci.yml]: do we need this?

**Comment by LucioFranco** [comment]
we should probably run this to make sure it all works, prob can do that via bors try?

**Comment by azriel91** [comment]
bors try

**Comment by LucioFranco** [comment]
That didnt actually test this :/

@torkleyy maybe we can try to enable gitlab for just this branch temporarily?

File: .gitlab-ci.yml (hunk 1 above)
Comment (torkleyy) [reply .gitlab-ci.yml]: Well, it‘s additional debug info. Just a habit to print version numbers ^^

**Comment by fhaynes** [comment]
Closing this as we are not using GitLab CI/CD atm

**Pull request closed without merging**